### PR TITLE
[CodeHealth] Avoid constant arrays of string pointers

### PR DESCRIPTION
**Pull request by cdesouza-chromium**
This PR changes several places where the construct constant arrays for string pointers are being (`constexpr const char* kValue ...`). This makes code overall safer. In most cases, a `std::array` of `string_view` is fine. However, in some places a fixed flat set is the better choice, as we are doing searches in some of these arrays.

Resolves

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/brave_ads/device_id/device_id_impl_linux.cc [hunk 1]
```diff
@@ -96,7 +96,7 @@ std::string GetDiskUuid() {
 
   // Look for first device name matching an entry of `kDeviceNames`.
   std::string result;
-  for (const char* device_name : kDeviceNames) {
+  for (auto device_name : kDeviceNames) {
```
Comment (aseren) [inline browser/brave_ads/device_id/device_id_impl_linux.cc:99]: nit: `const auto device_name : kDeviceNames`

File: browser/brave_ads/device_id/device_id_impl_linux.cc [hunk 2]
```diff
@@ -153,11 +151,9 @@ class MacAddressProcessor {
   std::string GetMacAddress() const { return mac_address_; }
 
  private:
-  bool IsValidPrefix(const char* name,
-                     const char* const prefixes[],
-                     size_t prefixes_count) {
-    for (size_t i = 0; i < prefixes_count; i++) {
-      if (UNSAFE_TODO(strncmp(prefixes[i], name, strlen(prefixes[i]))) == 0) {
+  bool IsValidPrefix(base::span<const char> name) const {
+    for (auto prefix : kNetDeviceNamePrefixes) {
```
Comment (aseren) [inline browser/brave_ads/device_id/device_id_impl_linux.cc:155]: nit: `const auto prefix : kNetDeviceNamePrefixes`

File: browser/net/brave_network_audit_test_helper.cc [hunk 3]
```diff
@@ -17,16 +20,13 @@ namespace brave {
...
-    "(::f{4}:)?10\\.([0-9]{1,3})\\.([0-9]{1,3})\\.([0-9]{1,3})",
-    "(::f{4}:)?192\\.168\\.([0-9]{1,3})\\.([0-9]{1,3})",
-    "(::f{4}:)?172\\.(1[6-9]|2\\d|30|31)\\.([0-9]{1,3})\\.([0-9]{1,3})",
-    "(::f{4}:)?127\\.([0-9]{1,3})\\.([0-9]{1,3})\\.([0-9]{1,3})",
-    "(::f{4}:)?169\\.254\\.([0-9]{1,3})\\.([0-9]{1,3})",
-    "f[cd][0-9a-f]{2}:.*",
-    "fe80:.*",
-    "::1",
-    "::"};
+inline constexpr auto kPrivateIPRegexps = std::to_array<std::string_view>(
```
Comment (aseren) [inline browser/net/brave_network_audit_test_helper.cc:23]: seems like `inline` is not needed here.

File: browser/ntp_background/ntp_p3a_helper_impl.cc [hunk 4]
```diff
@@ -40,8 +42,9 @@ constexpr char kNewTabPageEventCountConstellationDictPref[] =
     "brave.brave_ads.p3a.ntp_event_count_constellation";
 constexpr char kNewTabPageKnownCampaignsDictPref[] =
     "brave.brave_ads.p3a.ntp_known_campaigns";
-constexpr const char* kAllCreativeCountDicts[] = {
-    kNewTabPageEventCountDictPref, kNewTabPageEventCountConstellationDictPref};
+inline constexpr auto kAllCreativeCountDicts = std::to_array<std::string_view>(
```
Comment (aseren) [inline browser/ntp_background/ntp_p3a_helper_impl.cc:45]: seems like `inline` is not needed here.

File: browser/search_engines/search_engine_provider_util.cc [hunk 5]
```diff
@@ -27,6 +29,13 @@
 
 namespace brave {
 
+namespace {
+inline constexpr auto kTargetCountriesForEnableSearchSuggestionsByDefault =
```
Comment (aseren) [inline browser/search_engines/search_engine_provider_util.cc:33]: same comment for `inline` here.

**Review by aseren — APPROVED** [review]
LGTM, with some comments

File: browser/net/brave_network_audit_test_helper.cc (hunk 3 above)
Comment (cdesouza-chromium) [reply browser/net/brave_network_audit_test_helper.cc:23]: Done.

File: browser/ntp_background/ntp_p3a_helper_impl.cc (hunk 4 above)
Comment (cdesouza-chromium) [reply browser/ntp_background/ntp_p3a_helper_impl.cc:45]: Done.

File: browser/search_engines/search_engine_provider_util.cc (hunk 5 above)
Comment (cdesouza-chromium) [reply browser/search_engines/search_engine_provider_util.cc:33]: Done

**Review by stoletheminerals — APPROVED** [review]

**Review by iefremov — APPROVED** [review]

**Pull request merged**